src/app/App.js

```diff
@@ -1,68 +1,50 @@
-import React, { useEffect } from 'react';
-import { Switch, Route, Link } from 'react-router-dom';
-import { auth, createUserProfileDocument } from '../firebase/firebaseClient';
-import { setCurrentUser } from '../redux/user/user.actions';
-import { Provider } from 'react-redux';
-import { connect } from 'react-redux';
-import store from '../redux/store';
+import React, {useEffect} from 'react';
+import {Switch, Route, Redirect} from 'react-router-dom';
+import {authStateCheck} from '../redux/user/user.actions';
+import {connect} from 'react-redux';
 import Layout from '../layout/Layout';
-import Login from '../components/Login/Login';
+import SignIn from "../components/Auth/SignIn/SignIn";
+import SignUp from "../components/Auth/SignUp/SignUp";
 import './App.scss';
 
 const App = (props) => {
-  useEffect(() => {
-    auth.onAuthStateChanged(async (userAuth) => {
-      if (userAuth) {
-        const userRef = await createUserProfileDocument(userAuth);
+  const {isAuth, authStateCheck} = props;
 
-        userRef.onSnapshot((snapShot) => {
-          props.setCurrentUser({
-            id: snapShot.id,
-            ...snapShot.data(),
-          });
-        });
-      }
-      props.setCurrentUser(userAuth);
-    });
-  }, []);
+  useEffect(() => {
+    authStateCheck();
+  }, [authStateCheck]);

```

Comment:
tutaj uzywam zmiennej isAuth zeby sprawdzac cy user jest zalogowany nizej w mapDispatch masz pokazane jak to dziala + wywoluje metode authStateCheck czyli to co bylo wczesniej ale po prostu przenioslem to do akcji zeby byl porzadek